crates/nu-parser/src/parse_shape_specs.rs

```diff
@@ -82,7 +82,12 @@ pub fn parse_completer(
         ));
         return None;
     }
-    working_set.find_decl(cmd_name)
+
+    let decl_id = working_set.find_decl(cmd_name);
+    if decl_id.is_none() {
+        working_set.error(ParseError::UnknownCommand(span));
+    }
+    decl_id
```

Comment:
Maybe we want to move

https://github.com/nushell/nushell/blob/b226bf548f1bb2b84fc8415a7fefcc3f9951dc8f/crates/nu-parser/src/parser.rs#L4267

to this `parse_completer`, as we probably will add support of `@[]` syntax for `Completion:List`.